services/system/AuthSig/src/AuthSig.cpp

```diff
@@ -83,32 +84,24 @@ namespace SystemService
          authTable.put(AuthRecord{.account = getSender(), .pubkey = std::move(key)});
       }
 
-      void AuthSig::stagedAccept(uint32_t staged_tx_id, psibase::AccountNumber actor)
+      bool AuthSig::isAuthSys(psibase::AccountNumber              sender,
+                              std::vector<psibase::AccountNumber> authorizers)
       {
-         check(getSender() == StagedTxService::service, "can only be called by staged-tx");
+         auto accDetails = to<Accounts>().getAccount(sender);
+         check(accDetails.has_value() && accDetails->authService == AuthSig::service,
+               "sender " + sender.str() + " does not use this auth service");
```

Comment:
Is this check needed? `checkAuthSys` doesn't have it. If it doesn't hold, it's the caller's fault and the only harmful consequence is that the result is useless to the caller. I don't see it as being likely to happen by accident either, as the caller probably got this service in the first place by looking it up in accounts.

Reply:
>  the only harmful consequence is that the result is useless to the caller

Without this check, the call will silently seem to succeed, and the auth service might even reply `true`, causing the caller's bug to not only result in a failure, but in an actual erroneous execution of the staged tx. This could happen if the account previously used this auth service so it actually has a configuration.

So... it's not strictly necessary but I was thinking the cost of this check is worthwhile even though I agree it's an unlikely bug.

Reply:
The auth service is always looked up right before calling it, so I don't see how a previous auth service matters.

Reply:
> The auth service is always looked up right before calling it, 

The bug in the hypothetical disagrees with this by definition, otherwise there wouldn't be a bug. It's the caller's fault, I agree, but I was just considering how to minimize damage in this edge case. Perhaps it's just confusing, though. I'll remove it.

Reply:
The actual usage always does a lookup and immediate call, and that's by far the easiest and most natural way to use it. It's just a lot of extra code for such an unlikely error.

Reply:
removed